Skin layout revamp dev

ferranlala opened the pull request:
Summary of changes:
-Removed the layout option buttons of the top of LateNight
-Added layout options to view menu. Rearranged view menu.
-Added the option to hide the crossfader to LateNight and Shade
-Modified LateNight so that the library cannot be hidden when the skin is in medium or full size.

rryan committed:
Fix the build on OS X.

rryan committed:
Merge pull request #411 from uklotzde/NewSoundSourceAPI

New SoundSource/AudioSource API

ywwg committed:
LateNight: Fix effect knobs cut off on Mac

uklotzde committed:
Fix the Windows build (SoundSourceMediaFoundation)

jercaianu committed:
Redone implementation with better sound output

jercaianu committed:
Changed depth limits to 0 and 1

jercaianu committed:
improved sound quality

rryan committed:
Merge pull request #562 from uklotzde/SoundSourceMediaFoundationBuild

Fix Windows build (SoundSourceMediaFoundation)

daschuer committed:
mp3: Workaround for different sample rates in on file

daschuer committed:
mp3: made derPerFrameRate local

daschuer committed:
mp3: move switch case blocks to helper functions

rryan committed:
Add "stabilization read" to seeks for MP3s in SoundSourceCoreAudio.

SoundSourceProxyTest.seekForward was failing on OSX due to
cover-test.mp3 not producing accurate samples on a seek. It seems this
is related to the MP3 decoder not being "primed" right after a seek. To
compensate for this, we seek backwards 29 MP3 frames (same as in
SoundSourceMp3) and read forward.

The test passes now but it would be ideal if we had a reliable way to
determine the right amount to read on a per-file basis from ExtAudioFile
directly (in case other formats need pre-fetching). I'm following up
with Apple support for help here.

daschuer committed:
Merge pull request #564 from daschuer/mp3samplerate

mp3: Workaround for different sample rates in on file

rryan committed:
Merge pull request #565 from rryan/extaudiofile

Add "stabilization read" to MP3 seeks in SoundSourceCoreAudio.

jercaianu committed:
Rewrote phaser with much better sound output

uklotzde committed:
Fix handling of mono MP3 frames within a stereo file

uklotzde committed:
Abort loading of MP3 file on error before determining sampling rate

uklotzde committed:
Limit the maximum number of channels in MP3 files to 2

uklotzde committed:
Fix debug assertion to accept MP3 files without any tags

uklotzde committed:
Some minor code reorderings when loading MP3 files

uklotzde committed:
Fix libMAD error handling for MP3 decoding

The decoding functions do NOT return the actual error code!! Instead
the error code must be read from mad_stream if the decoding functions
returned != 0.

This caused some really ugly sounds at the beginning of some MP3 files,
because frames were not skipped as suppossed to be.

uklotzde committed:
Separate code paths for reading mono/stereo MP3 data

Slightly more code, but hopefully easier to understand.

daschuer committed:
Merge pull request #568 from uklotzde/MP3ChannelAndDecodingFix

MP3 channel and decoding fix

uklotzde committed:
Fix Coverity warning for SoundSourceSndFile

*** CID 61970:  Uninitialized members  (UNINIT_CTOR)
/src/sources/soundsourcesndfile.cpp: 17 in Mixxx::SoundSourceSndFile::SoundSourceSndFile(QUrl)()
11         return list;
12     }
13
14     SoundSourceSndFile::SoundSourceSndFile(QUrl url)
15             : SoundSource(url),
16               m_pSndFile(NULL) {
>>>     CID 61970:  Uninitialized members  (UNINIT_CTOR)
>>>     Non-static class member field "m_sfInfo.seekable" is not initialized in this constructor nor in any functions that it calls.
17     }
18
19     SoundSourceSndFile::~SoundSourceSndFile() {
20         close();
21     }

uklotzde committed:
Fix Coverity warning for SoundSourceMp3

*** CID 61968:  Uninitialized members  (UNINIT_CTOR)
/src/sources/soundsourcemp3.cpp: 163 in Mixxx::SoundSourceMp3::SoundSourceMp3(QUrl)()
157               m_fileSize(0),
158               m_pFileData(NULL),
159               m_avgSeekFrameCount(0),
160               m_curFrameIndex(kFrameIndexMin),
161               m_madSynthCount(0) {
162         m_seekFrameList.reserve(kSeekFrameListCapacity);
>>>     CID 61968:  Uninitialized members  (UNINIT_CTOR)
>>>     Non-static class member field "m_madSynth.pcm" is not initialized in this constructor nor in any functions that it calls.
163     }
164
165     SoundSourceMp3::~SoundSourceMp3() {
166         close();
167     }

uklotzde committed:
Eliminate unneeded member variable in SoundSourceSndFile

uklotzde committed:
Symmetric invocation of init()/finish() functions in SoundSourceMp3

jercaianu committed:
Added stereo option and removed redundant coefficient computing

rryan committed:
Merge pull request #569 from uklotzde/FixCoverityWarningsForSoundSources

Fix coverity warnings for new SoundSources

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Do not segfault when writing tags into files with inaccessible tags

Observed for some .m4a files when TagLib failed to access (read/write) the
existing MP4 tag.

demos committed:
missing coef

demos committed:
Merge branch 'pan_effect_stereo' of https://github.com/demos/mixxx into pan_effect_stereo

jercaianu committed:
Phaser now has only even number of stages

uklotzde committed:
Improve readability of tag writing code

daschuer committed:
Merge pull request #571 from uklotzde/TagWritingFixes

Writing Tags: Handle inaccessible tags correctly

demos committed:
beginning of period unit switch

jercaianu committed:
Tweaked some default values and added better names for variables

jercaianu committed:
Reformated sample processing part

default-kramer committed:
fix Windows build

rryan committed:
Version bump master to avoid confusion with 1.12 branch.

rryan committed:
Merge pull request #572 from default-kramer/fixWindowsBuild

fix Windows build

jercaianu committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into phaser

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

demos committed:
the delay moves with the sound

demos committed:
cleaning

daschuer committed:
Coverity Scan: Removed dead code

daschuer committed:
Fix error in CachingReader::read that prevent seeking into preroll, fixes Bug #1450260

demos committed:
cleaning + better texts + delay debug

uklotzde committed:
Rename min/max frame index functions to improve readability

uklotzde committed:
Cleanup AudioSource API and use consistent naming

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Add note about SIGBUS errors if MP3 file becomes inaccessible

https://bugs.launchpad.net/mixxx/+bug/1452005

demos committed:
changing the scale of the period knob

demos committed:
changing the scale of the period knob

demos committed:
logarithmic smoothing + cleaning

uklotzde committed:
Avoid unnecessary integer casts

uklotzde committed:
Replace "magic numbers" with constants

uklotzde committed:
Reduce visibility of constants and functions in AudioSource

uklotzde committed:
Move MP3 specific constant into separate header file

uklotzde committed:
Remove redundant static qualifier from constant in anonymous namespace

rryan committed:
Merge pull request #577 from uklotzde/AudioSourceApiCleanup

AudioSource API Code Cleanup

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

uklotzde committed:
Log warning about unreadable tags in audio files

uklotzde committed:
Change parameter type of getSoundSource() from QString to QUrl

daschuer committed:
enginefilterpane: use linear stetching instead of ramping

daschuer committed:
add enginefilterpanesingle

daschuer committed:
Used enginefilterpanesingle in auto pane effect

daschuer committed:
Use ramped value for delay

jercaianu committed:
Merge branch 'master' of github.com:mixxxdj/mixxx into phaser

ywwg committed:
Merge branch '1.12'

Conflicts:
	src/test/mixxxtest.cpp

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

uklotzde committed:
Add an assertion to detect invalid conversions QString <-> QUrl

uklotzde committed:
New streamlined SoundSource plugin API version 8

A simple C++ API that reduces the chance of memory leaks that were
imminent in the previous C/C++ version. Error handling and logging
in SoundSourceProxy while loading plugins has been improved.

uklotzde committed:
Allow CachingReaderWorker to return partially read chunks

This might happen for corrupt/truncated files. Currently there is no
special case handling in CachingReader. Partially (= incomplete) read
chunks are simply treated like successfully (= complete) read chunks.

Without this fix the CachingReader tried to read the remaining samples
in the last chunk of a corrupt MP3 file over and over.

uklotzde committed:
Detach dynamic library management from SoundSourceProxy

jercaianu committed:
Fixed bubbling noise from the 'Rate' knob, added tanh clipper and comments

uklotzde committed:
Introduce SoundSourceProvider interface

SoundSourceProvider is a generic interface that will implemented both by
plugin libraries as well as built-in sound sources. This will simplify
the selection of an appropriate SoundSource in SoundSourceProxy and is
useful for future extensions like selecting one out of multiple providers
for each file type.

uklotzde committed:
Add and register a SoundSourceProvider for each SoundSource

uklotzde committed:
Fix FFmpeg build

uklotzde committed:
Give SoundSource providers a name

jercaianu committed:
added tanh

uklotzde committed:
Delete obsolete #include directive

jercaianu committed:
clipped only feedback

jercaianu committed:
updated frequency description

uklotzde committed:
Wrap static members from SoundSourceProxy into class

uklotzde committed:
Fix static initialization of SoundSource providers in tests

uklotzde committed:
Fix SoundSource plugin warning

uklotzde committed:
Log detailed debug info about SoundSource providers

uklotzde committed:
Move final debug logging out of loop

uklotzde committed:
Delete obsolete member from SoundSourceProxy

uklotzde committed:
Avoid unnecessary locking in SoundSourceProxy

uklotzde committed:
Eliminate duplicate initialization in tests

uklotzde committed:
Add debug logging for seek test

jercaianu committed:
Fixed comment indentation and rate knob issue

jercaianu committed:
Added linear knob until toggle one is fixed

uklotzde committed:
Simplify SoundSource plugin registration

demos committed:
Merge branch 'master' of git://github.com/mixxxdj/mixxx into pan_effect_stereo

demos committed:
adding comments to filterpansingle

Commit:
Changed RateRange to RateRangeDecimal

demos committed:
adding comments to filterpansingle 2

demos committed:
Merge pull request #2 from daschuer/pan_effect_stereo

pan delay per sample with linear stretching

demos committed:
Merge branch 'daschuer_filterpansingle_for_autopan' of https://github.com/demos/mixxx into pan_effect_stereo

demos committed:
Merge branch 'pan_effect_stereo' of https://github.com/demos/mixxx into pan_effect_stereo

daschuer committed:
Merge pull request #586 from uklotzde/CachingReaderPartialChunkRead

CachingReaderWorker: Don't discard partially read chunks

uklotzde committed:
Delete obsolete member functions and code cleanup

uklotzde committed:
Use plugin file name filter only for Apple OS X

At least this is how it works in 1.12.

Commit:
Keep ComboBoxRateRange and add some more sane values, plus allow any
setting in the config file

Commit:
Limit Rate to 90% max and better labels

Commit:
Surround * and / with spaces

Commit:
Name new config RateRangePercent and rename slot/variables accordingly

daschuer committed:
Merge pull request #560 from jercaianu/phaser

phaser effect

uklotzde committed:
Change parameter name: fileName -> libFilePath

uklotzde committed:
Change getter name: getFileName() -> getFilePath()

The corresponding Qt function name is actually misleading.

uklotzde committed:
Change the name of the MP3 SoundSource provider

uklotzde committed:
Add a TODO note about network shares

According to the documentation of QUrl::isLocalFile()/toLocalFile()
network shares are handled in the same way as before, i.e. they are
considered to be local files with all consequences.

uklotzde committed:
Filtering of plugin file names on UNIX systems (not only OS X)

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

demos committed:
pausable period

daschuer committed:
Merge pull request #588 from MK-42/rateAsDecimal

Changed RateRange to RateRangeDecimal

ywwg committed:
Minor style and English language fixes

demos committed:
position of the sound fixed when period knob is used

demos committed:
period will not be fixed anymore if the effect is in enabling state or the position change is ramped

demos committed:
period will not be fixed anymore if the effect is in enabling state or the position change is ramped

demos committed:
reintroducing lawcoef : *2 at center and * 1.7 at left or right

uklotzde committed:
Streamline the MetadataSource interface

Usually both metadata and cover art should be parsed at once. Optional
parameters are used if only one or the other is needed. A single
virtual function covers all those use cases.

uklotzde committed:
Delete obsolete member function from TrackDAO

uklotzde committed:
Add checks for supported file types to SoundSourceProxy

uklotzde committed:
Remove obsolete local variables

uklotzde committed:
Consistent naming: "file type" instead of "file extension"

uklotzde committed:
Delete obsolete functions from SoundSourceFFmpeg

daschuer committed:
Merge pull request #460 from demos/pan_effect_stereo

Pan effect

Commit:
Highlight Crates a Track is in. Still some work to do

Commit:
Rename delegate according to his function

uklotzde committed:
Fix indentation

uklotzde committed:
Fix and comment parsing of cover art

uklotzde committed:
Restore deleted comment

uklotzde committed:
Improve comment in MetadataSource interface

uklotzde committed:
Add and improve functions for backward compatibility

Commit:
Highlighting crates the selected track is in works now

Commit:
Some minor code-style fixes

Commit:
Added comment to CrateHighlightDelegate::initStyleOption

Commit:
Added highlighting of playlists

Commit:
Another aproach for Repainting preventing the selection to move

rryan committed:
Ignore QModelIndex's that don't have TreeItem internal pointers in CrateHighlightDelegate.

rryan committed:
Fix SidebarModel handling of child models' dataChanged signals.

MK-42 committed:
Merge pull request #1 from rryan/sidebar-highlight

Sidebar highlight fixes.

Commit:
Holding Track-Crate relation in memory for faster highlighting

Commit:
Holding Track-Playlist relation in memory for faster highlighting

Commit:
Clean up used memory

Commit:
fix missing signals

Commit:
Clear selected Playlists/Crates on viewChange

rryan committed:
Remove CrateHighlightDelegate.

rryan committed:
Add bold attribute to TreeItem.

rryan committed:
Restore LibraryFeature::trackSelected signal.

I think this was why we had to add signals to each individual feature.

rryan committed:
Respond to Qt::FontRole in SidebarModel.

rryan committed:
Populate playlist and crate TreeItems with bold attribute when track selection changes.

rryan committed:
Populate TreeItem state when re-building the child model.

This supports bolding/unbolding the playlist/crate when a track is added/removed.

MK-42 committed:
Merge pull request #2 from rryan/sidebar-highlight

Some fixes.

rryan committed:
Swap key/value pairing of crate/playlist membership cache in CrateDAO/PlaylistDAO.

Also add a method to fetch all the playlists/crates a track is in.

uklotzde committed:
Correctly distinguish between "file extension" and "SoundSource type"

rryan committed:
Fetch playlist / crate membership set on track selection.

Avoids repeatedly asking the DAO for pair-wise membership (requires
linear hash table lookups).

uklotzde committed:
Initialize FFmpeg before registering providers and loading plugins

uklotzde committed:
Delete redundant DEBUG_ASSERT

Empty file names are handled gracefully by SoundSourceProxy.

rryan committed:
Fix type mismatch.

rryan committed:
Add move constructor / move assignment operator for SampleBuffer.

Fixes the build for -std=c++0x.

rryan committed:
Enable -std=c++11 support.

rryan committed:
Update Travis to use gcc 4.8 for c++11 support.

uklotzde committed:
Restore comment about QApplication in tests

uklotzde committed:
Reorder and add missing #include directives

uklotzde committed:
Static allocation of SoundSourceProvider in plugins

uklotzde committed:
Symmetrically allocate and delete SoundSource from within plugins

uklotzde committed:
Use nullptr (C++11) instead instead of NULL (deprecated)

uklotzde committed:
Fix move constructor in SampleBuffer

rryan committed:
Merge pull request #604 from uklotzde/SampleBufferMoveConstructorFix

Fix move constructor of SampleBuffer

uklotzde committed:
Merge branch 'master' into SoundSourcePluginUrlFileNameFix

uklotzde committed:
Use C+11 override keyword

uklotzde committed:
Add comments about singleton/shared SoundSourceProvider instances

uklotzde committed:
Unify export of SoundSources from plugin libraries

uklotzde committed:
Prefer local static variables for shared data in plugins

rryan committed:
Merge pull request #578 from uklotzde/SoundSourcePluginUrlFileNameFix

SoundSource Plugin API: Version 8

rryan committed:
Fix c++11 TODO.

uklotzde committed:
Fix memory leak in NotNode

uklotzde committed:
Add (redundant) keyword 'virtual' to declaration of destructor

rryan committed:
Add public domain std::make_unique shim from Brandon Streiff.

Taken from: https://github.com/bstreiff/cppbits

Thank you @bstreiff!

uklotzde committed:
Replace plain pointers with std::unique_ptr to avoid memory leaks

uklotzde committed:
Delete obsolete constructors and destructors

uklotzde committed:
Add 'override' keyword to inherited virtual functions in derived classes

uklotzde committed:
Fix (or comment) fragile design of Numeric-/DurationFilterNode

Virtual member functions should not be called from within the constructor!

uklotzde committed:
Consistent wrapping of SQL expressions into parantheses

uklotzde committed:
Improve readability using auto and std::make_unique

uklotzde committed:
Don't use auto for derived class ptr to fix compile error in CI build

uklotzde committed:
Revert previous changes and try again to fix CI build errors

rryan committed:
Remove QColor::darker call.

Since it does not mutate its instance it has no effect.

rryan committed:
Remove no-args constructor parentheses.

rryan committed:
Use QDomNodeList::count instead of length.

In Qt5 QDomNodeList::length changed from unsigned int to int. To avoid
compiler warnings when using one or the other use count.

rryan committed:
Move QtScript ByteArrayClass into lib/.

rryan committed:
Move reverb plugin to lib/reverb.

rryan committed:
Remove some instances of FALSE and TRUE.

rryan committed:
Remove use of deprecated (in c++11) register keyword.

The generated assembly with and without is identical in EngineFilterBessel4/8.

rryan committed:
Fix crate/playlist membership fetchers.

rryan committed:
Don't call TreeItemModel::setData in a loop.

This caused severe performance issues since it triggered a repaint on every loop
iteration.

rryan committed:
Merge pull request #617 from rryan/sidebar-highlight

Highlight Crates a Track is in

rryan committed:
Merge pull request #614 from uklotzde/QueryNodeRefactoring

QueryNode: Memory leak fix + refactoring

uklotzde committed:
CachingReader: Always read complete chunks

Fixes the following assertion:
DEBUG ASSERT: "0 == (hint.sample % CachingReaderWorker::kChunkChannels)" in file src/cachingreader.cpp, line 492

The calculation of chunk offsets relies on chunks with a constant size.
Only the last chunk might be smaller than all others. The worker now fills
the chunks with silence if the data has been read partially.

uklotzde committed:
CachingŔeaderWorker: Don't seek beyond EOF

Because of reasons I don't know processChunkReadRequest() is called
periodically even while the track is not playing. Unnecessary seek
operations on the SoundSource should be avoided for if the frame index
is out of range or beyond the end of the file.

uklotzde committed:
Handle trivial/edge cases in SoundSourceMp3 separately to avoid prefetching

uklotzde committed:
Align seeking in SoundSourceM4A with SoundSourceMp3 to avoid prefetching

uklotzde committed:
Merge branch '1.12' of https://github.com/mixxxdj/mixxx.git into Merge1.12ToMaster

# Conflicts:
#	src/library/searchquery.h
#	src/library/searchquery.cpp
#	src/main.cpp
#	src/soundsourcemodplug.cpp
#	src/test/coverartcache_test.cpp
#	src/test/soundproxy_test.cpp

daschuer committed:
Merge pull request #628 from uklotzde/Merge1.12ToMaster

Resolve merge conflicts from 1.12 to master

uklotzde committed:
Merge branch 'master' into CachingReaderAvoidUnnecessarySeeking

uklotzde committed:
Improve error dectection when seeking corrupt MP3/M4A files

uklotzde committed:
Fix wrong assert in SoundSourceM4A

uklotzde committed:
No need to overwrite the result value from the MP3 decoder

ferranlala committed:
LateNight: crossfader can now be hidden.
Added "Show crossfader" check option to view menu.

ferranlala committed:
Fixed double border when crossfader was hidden.

ferranlala committed:
Removed layout option buttons from the top of LateNight.
Centered the clock.

ferranlala committed:
Reordered view menu.

daschuer committed:
Merge remote-tracking branch 'upstream/1.12'

uklotzde committed:
Replace NULL with nullptr

uklotzde committed:
Merge branch 'master' into CachingReaderAvoidUnnecessarySeeking

uklotzde committed:
Add missing nullptr check

daschuer committed:
Merge pull request #626 from uklotzde/CachingReaderAvoidUnnecessarySeeking

Fix outstanding SoundSource issues

ferranlala committed:
Added "Show Library" check control to the view menu.

ferranlala committed:
Changed CO show_library_tab_small to show_library in LateNight.
This lets the view menu check option work in all circumstances.

ferranlala committed:
Added Show Eqs check control to the view Menu.

ferranlala committed:
Added Show Mixer check control in view menu.

ferranlala committed:
Added Show 4 decks check option to view menu.

ferranlala committed:
Fixed Maximize Library skin button not updating the view menu check option.

ferranlala committed:
Updated keyboard shortcuts for layout options.

ferranlala committed:
Added show crossfader option the Shade.

ferranlala committed:
LateNight: Library always visible when skin is med or full size.
The library has been disconnected from show_library CO when the lower half of the skin is either in full or med mode.
This makes sense since there's always enough space for the library in both modes, even if the samplers, the fx chains and/or the mics are visible.
The library has been changed to minimum expanding height to accomodate to the height of visible sections samplers, fx and/or mic.
These changes also solve the library not showing up on Mixxx start if the samplers, fx or mics were selected on the small mode when Mixxx exit.

ywwg commented:
I'll try to take a look at this later, thanks

Be-ing commented:
Nice, this feels better. A few small issues I notice in LateNight are that the "W" at the end of "PREVIEW" in the preview deck is cut off and there are no tall waveforms anymore.
![mixxx-latenight-pr639](https://cloud.githubusercontent.com/assets/9455094/8552025/e4b63a34-249f-11e5-9f75-d8d25b450f79.png)

Be-ing commented:
With the skin option buttons removed from the top, the clock looks a bit lonely up there. Maybe the clock could be aligned to the left of that bar and the Mixxx logo centered?

daschuer commented on res/skins/LateNight/mixer.xml:
```diff
@@ -147,6 +147,10 @@
+                            <Connection>
+						<ConfigKey>[Master],show_xfader</ConfigKey>
+							<BindProperty>visible</BindProperty>
+						</Connection>
```
Nitpick: replace tabs with with four spaces.

daschuer commented on res/skins/Shade/mixer_panel.xml:
```diff
@@ -137,6 +137,10 @@
+                            <Connection>
+                                <ConfigKey>[Master],show_xfader</ConfigKey>
+                                <BindProperty>visible</BindProperty>
+                            </Connection>
```
Blame on me, for the bad alignment.

daschuer commented:
What is the use-case for removing the xFader in Shade. Any idea how to use the empty area? 
This will allow a Numark NDX three button laylout or something fancy like vinyl break effects.

ferranpujolcamins commented:
In the initial discussion, Owen replied:

"If you add "hide crossfader" to the menu you'll also have to add support
for hidden
crossfader to Deere and Shade as well. Otherwise people using those
skins will file bugs saying they tried to hide the crossfader and it's
not working."

So I've just pleased him :)

We could just make the faders and the meters longer.

2015-07-07 20:31 GMT+02:00 Daniel Schürmann <notifications@github.com>:

> What is the use-case for removing the xFader in Shade. Any idea how to use
> the empty area?
> This will allow a Numark NDX three button laylout or something fancy like
> vinyl break effects.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/639#issuecomment-119294358>.
>

daschuer commented:
LateNight: (This is all a matter of taste. @ywwg will have the final word) 
* The Clock bar is not aligned with the empty bar on top of the decks. 
* On my Cinnamon desktop the skin comes up with the layout buttons below my display. How about move the footer onto the top and merge it with the clock and the CPU widget? 
* Why did you remove the 4/2 Decks button? Is this more a config option instead of a live feature? 
* What is the use-case for hide the EQs? The saved screen seams to be unused. 
* The preview deck looks sonhow foreign. Can we integrate it to the Mixxer area? Big Mixer = Small preview deck and vice verse.

daschuer commented:
@ferranpujolcamins: Thank you for working on this. IMHO with some tweaks it suits to be part of 1.12.
How about the additional strings? Will this be a problem since we are already past beta?

ferranpujolcamins commented:
-I already had planned to reduce the minimum height of the skin (it will be in another PR). @daschuer could you please tell me which is your screen resolution? Also, could you find out which is the maximum "minimum height" of the skin that allows you to maximize Mixxx in Cinnamon?

-I personally think the 2/4 decks button is something you set up once and don't touch during a performance. If someone heavily uses it I don't mind to add the button to the bar.

-The show/hide EQ was already there before this PR.

-The preview deck can surely be improved, in general it is our duty now to improve the skins to take advantage of every bit of space a hidden feature leaves, but this is out of scope of this PR.

- @Be-ing what do you mean by "tall waveforms"? When the mixer is hidden the waveforms appear a bit taller, just like in trunk. So nothing has changed here. Also, I can't reproduce the text issue with the preview deck.

ferranlala committed:
Tabs->Spaces. Minor formatting fix.

ferranpujolcamins commented:
I've fixed the formatting issues.
I'm investigating what can be done with the clock and the latency meter. I'll send you screenshots.

Be-ing commented:
> @Be-ing what do you mean by "tall waveforms"? When the mixer is hidden the waveforms appear a bit taller, just like in trunk. So nothing has changed here.

You're right. Sorry, I got confused because the 1.12 branch currently has the option to toggle big/small waveforms independent of showing the mixer (which I think is kinda awkward and unnecessary).

> Also, I can't reproduce the text issue with the preview deck.

I am using KDE Plasma 5.3 on Fedora 22.

Be-ing commented:
Alternatively, the clock and CPU meter could be moved near the Mixxx logo. I suspect it would look better if they stay up top though.

ywwg commented:
> currently has the option to toggle big/small

this was removed a while ago, you might have to update to see the difference.

ferranpujolcamins commented:
Can't we get rid of the Mixxx logo?

ferranpujolcamins commented:
What about this?
![captura de pantalla de 2015-07-07 23 13 00](https://cloud.githubusercontent.com/assets/6429775/8557445/bb18fd02-24fd-11e5-896d-ae6cd4090b47.png)

Be-ing commented:
Hmm, it could be a little weird/confusing having the horizontal CPU meter adjacent to the master VU meter. What if the clock and CPU meter were below the Mixxx logo?

daschuer commented:
Mixxx logo: See the discussion here:  
https://github.com/mixxxdj/mixxx/commit/0c6c6ffb1cfae7e993d71620eee54c4a82032470#commitcomment-10058233

Maximize is not possible on my 1366x786 cinnamon screen. 
I use one head panel with the size of 25 pixel. Header bar and tool bar are also 25 Pixel.
So I think the maximum skin height 736 for me.

ferranpujolcamins commented:
I don't really like this way.
![captura de pantalla de 2015-07-07 23 17 58](https://cloud.githubusercontent.com/assets/6429775/8557567/80568c9c-24fe-11e5-8683-efa7bfda1053.png)

Be-ing commented:
Yeah, that is pretty odd now that I can see it. What about putting the clock in the top right of that section and the CPU meter in the top left?

ferranpujolcamins commented:
So @daschuer would LateNight maximize in a "stock" Cinnamon desktop?

ferranpujolcamins commented:
I like this :)
![captura de pantalla de 2015-07-07 23 34 20](https://cloud.githubusercontent.com/assets/6429775/8557907/c5988cb8-2500-11e5-9003-e1b21e260ac2.png)

Even though the master meters might benefit from a little of padding on their right. However there's another PR that modifies their placement, so we shall wait for it to be applied before we make more tweaks to this.

Be-ing commented:
Yeah, that's better. I'm not sure if the horizontal divider between the clock and master VU meters is necessary.

ferranpujolcamins commented:
I don't really care.
![captura de pantalla de 2015-07-07 23 39 49](https://cloud.githubusercontent.com/assets/6429775/8557999/949d073c-2501-11e5-8a47-ede14df57002.png)

I'll wait to see what other people say and commit this.

ywwg commented:
> Can't we get rid of the Mixxx logo?

absolutely not

ywwg commented:
I think this is a really good start!  How did you get the view options to gray out on the Deere skin?

* Show Spinnies disappeared.  We have users who report much better performance on extremely crappy hardware if they can disable spinnies.  That should be one of the View options.  (note that the show spinny icon is still in each deck, but that's horrible and we need to change that toggle switch to something else)
* I really think 2/4 decks needs to be on the skin itself.  I find myself switching that setting a lot, even during a set.
* Same with preview deck
* Maybe put the clock and CPU meter in the bottom button bar?  Also maybe label the cpu meter with "CPU" (even though it's not technically CPU usage, it's more understandable than "load" or "latency"
* When turning off crossfader and eqs, it would be nice if the mixer was at the top of the window, not the bottom.

Be-ing commented:
> Maybe put the clock and CPU meter in the bottom button bar? Also maybe label the cpu meter with "CPU" (even though it's not technically CPU usage, it's more understandable than "load" or "latency"

That seems like a better idea than the last screenshot.

ferranlala committed:
Moved clock and latency meter to bottom bar. Moved Big library button to the left.

ferranlala committed:
Removed top toolbar.

ferranlala committed:
Unified all LateNight toolbars.

ferranlala committed:
Added the missing show library button to the bottom bar.

ferranlala committed:
LateNight: Added 2/4 decks button to bottom bar.

ferranlala committed:
Added show preview deck button to the bottom bar.

ferranlala committed:
LateNight: Recentered the clock.

ferranlala committed:
LateNIght: Added CPU label to the latency meter.

ferranlala committed:
LateNight: makes the mixer align on top.

ferranpujolcamins commented:
Ok, I've moved the clock and the latency meter to the bottom bar and added the 2/4 decks and the show preview deck buttons. I've also added a "CPU" label to the latency meter. I've also unified the different bottom bars that existed because they were almost identical anyway.

I've changed the mixer alignment to top, in general it looks nice to me. However, this makes the background of the crossfader too small. There's already a PR (https://github.com/mixxxdj/mixxx/pull/634) that removes the crossfader border and background, which I think is the way to go. Shall I merge this branch?

I agree about the show spinnies button. I've thought we could change that with a crossfader assign switch, like the ones there are already in the samplers.

Deere and Shade can hide each spinny individually. This means there will be 4 "show spinny on deck N" view menu options. However, LateNight handles all the spinnies at the same time with a special CO. How do we handle this? Right now the view menu options can't change according to which skin is loaded, isn't it? A few days ago I attempted to make the spinnies in LateNight individually "hiddable". However this led to the waveforms playing position not being aligned between decks with the spinny visible and decks with the spinny hidden. What's the way to go here? 

By the way, there's one problem: when on med or full mode, the show library CO does nothing. It behaves like this for two reasons:
1-It didn't make sense to be able to hide the library if the space was to be unused anyway.
2-The show library CO is a trigger for a widget stack when LateNight is in small mode.
If the last time LateNight was in small mode it had the sampler page selected (or fx or mic page), on the next Mixxx run, the show library CO is set to 0 because the selected tab is another one. This happens even if LateNight is in med mode, because the fx_samplers_container_small template is hidden but is still there. What this caused is that if LateNight was in med or full mode on Mixxx start, the library was hidden even if it was visible on the last Mixxx run. Disconnecting the library from the show library CO in med and full mode makes this bug disappear.

ferranpujolcamins commented:
PS: Maybe we could move the bar to the top of the skin. What do you think?

daschuer commented:
> There's already a PR (#634) that removes the crossfader border and background, which I think is the way to go. Shall I merge this branch?

No, lets avoid criss cross merge. We also need the permission to merge it and a clean rebase. 
Is https://github.com/mixxxdj/mixxx/pull/634 polished for merge? I will ask @synthor.

daschuer commented:
> PS: Maybe we could move the bar to the top of the skin. What do you think?

+1 this will solve issues on small screens. I like the clock centered on top

ywwg commented:
> PS: Maybe we could move the bar to the top of the skin. What do you think?

I'd prefer to keep it on the bottom for now.  I'll take a look at the clock placement and see if it's working.



> However, LateNight handles all the spinnies at the same time with a special CO. How do we handle this?

personally I don't see a need for spinny toggling on individual decks.  It doesn't really make sense as a per-deck selection.  So I'd prefer to adapt all of the skins to act like LateNight in that regard.

ferranlala committed:
Added show spinnies check option to the view menu.

ferranlala committed:
Reordered view menu.

ferranlala committed:
Updated keyboard shortcuts. The coverart shortcut is broken for all locales except spanish, russian and italian.

ferranlala committed:
LateNight: Changed show spinny button for crossfader assign button.
Swapped it by quantize button so the layout is like on the sampler decks.

ferranlala committed:
Shade: switched to common show/hide control for both spinnies.
Removed the show/hide spinny deck button and replaced for xFader assign.
Swaped position of xFader assign button and quantize button.

ferranpujolcamins commented:
Ok, this should be all.

Current issues:

-I couldn't correctly map the keyboard shortcuts for show/hide cover art. I don't know how to map the key next to the - (US layout). Right know I could only make it work with Spanish, Italian and Russian.

-PR 634 needs to be merged for the crossfader to look good in LateNight.

-We still have the "sometime, the show library button on LateNight does nothing" issue.

-I've deliberately not updated Deere because it needs major rework, so we will apply all this improvements to it when we put some love to it in the near future.

-Shade needs some love. Deere needs a whole lot of love. I'll probably put some time on them now that I'm finally on holidays. I thought I could make new cool graphics for Deere, something similar (but not equal) to this:
http://medialoot.com/blog/how-to-create-a-detailed-audio-rotary-knob-control-in-photoshop-illustrator/
Does the knobs really need to be vector graphics? If they are bitmaps it is much easier to make them look fantastic, like on the link.

-Something else?

synthor commented:
Wasn't able to repair my branch. So I created a new one of 1.12 and cherry-picked my four commits on it. See #641

Be-ing commented:
This is looking better and better. :) I like the replacement of the spinny toggle buttons with crossfader assignment buttons. This is a better use of the space now that Mixxx supports 4 decks.

I am in favor of moving the bar to the top. I also have issues with the buttons being cut off on my laptop screen. Before, I thought that was a bug in KWin 5, not Mixxx.
I think it would help to put a little spacer between the "LIBRARY" and "MIC/AUX" buttons so the "MIC/AUX", "EFFECTS", and "SAMPLERS" buttons appear as a group.

>I couldn't correctly map the keyboard shortcuts for show/hide cover art. I don't know how to map the key next to the - (US layout). Right know I could only make it work with Spanish, Italian and Russian

Ctrl and = is working for me on my en_US keyboard.

> We still have the "sometime, the show library button on LateNight does nothing" issue.

There also still https://bugs.launchpad.net/mixxx/+bug/1462061 which is really annoying.

ferranlala committed:
LateNight: Added dividers to the bottom bar.

ferranpujolcamins commented:
I really had no idea what I was doing when I set the keyboards shortcuts, so we really need someone to confirm they work as expected.

> I also have issues with the buttons being cut off on my laptop screen.
I'm working towards the reduction of the minimum height of LateNight. A first step is building a slider that can be resized, so I could borrow a few more pixels from the pitch slider. I've asked for help about this in the mailing list so if you know something about how the skinning engine behaves....:)

@ywwg , of all the issues we've mentioned in the discussion so far, which ones are blocking the merge of the PR? I want to start to work on other non-related skin improvements.

synthor commented:
captura de pantalla de 2015-07-07 23 39 49

> I'll wait to see what other people say and commit this.

In 4ch mode it would be a problem with the master meter. I placed it centered below the master knobs. Exactly in row with the channel meters.

daschuer commented:
> -I've deliberately not updated Deere because it needs major rework, so we will apply all this improvements to it when we put some love to it in the near future.

@esbrandt (the maintainer of Deere).  Can you give a status update? 

> -Shade needs some love. 

Yes. Do you have a special plan for it? 

> Deere needs a whole lot of love. I'll probably put some time on them now that I'm finally on holidays. I thought I could make new cool graphics for Deere, something similar (but not equal) to this:
http://medialoot.com/blog/how-to-create-a-detailed-audio-rotary-knob-control-in-photoshop-illustrator/

@esbrandt: (please correct me if I am wrong, I'm no design expert.) 
IMHO the original Deere idea was to use a flat-design for it. Recently we have introduced not that flat siders, which breaks the pure flat-design for me. 
In any case, I like the idea to have a flat-design pre-installed.

Your button looks more like skeuomorphism. In the German wikipedia are two nice pictures comparing the two designs: https://de.wikipedia.org/wiki/Flat_Design
I like the idea to offer a skeuomorphism design as well. So if you can effort the time, feel to fork any skin and add your own flavour to it. 

The 1.11 Deere design is for me also a candidate to lift to 1.12. 

> Does the knobs really need to be vector graphics? If they are bitmaps it is much easier to make them look fantastic, like on the link.

Mixxx 1.12 supports svg and png buttons. 

> -Something else?

Do you have an idea to solve the pending effect issue? We have requests to allow chaining filter end echo effect using LateNight.

synthor commented:
Yeah, the next thing I like to work on is the staticness of the effectchains in LateNight (and other skins also). Chains are pretty useless, when the artist can not configure them like he want and is limited in his creativity.

daschuer commented:
Four decks + Mixer + EQs + Preview Decks: 
If you enable the xFader, there is a ugly black area below the xfader. 
Without the xfader and the preview deck, the Mixer ends aligned with the decks. It would be nice if we could keep the alignement, and just add the xFader below.

synthor commented:
My commits would look like this:

![lnsr](https://cloud.githubusercontent.com/assets/13002524/8581428/ed390f70-25c0-11e5-8b58-1f3c036ebc9d.png)

ferranpujolcamins commented:
> If you enable the xFader, there is a ugly black area below the xfader.

Yeah synthor's commits fixes this.

> Without the xfader and the preview deck, the Mixer ends aligned with the decks. It would be nice if we could keep the alignement, and just add the xFader below.

I take note of this, but I'll fix that in another PR, this PR is not a major LateNight rework.

synthor commented:
> Yeah synthor's commits fixes this.

:)

ywwg commented:
This looks really good!  Make sure to pull in 1.12 so you get the crossfader update.

The only bug I see is that In the larger sizes, the library button is clickable but doesn't do anything.

I think the clock is fine where it is, but it might also work in the right corner along with CPU.  Up to you.

ywwg commented:
> staticness of the effectchains in LateNight (and other skins also).

One option is to make a generic FX chain template that can be applied to all skins with minor color theming.

ywwg commented:
> I want to start to work on other non-related skin improvements.

I think this set of improvements is fine for this PR.  we don't have to solve all the problems now

Be-ing commented:
> One option is to make a generic FX chain template that can be applied to all skins with minor color theming.

This seems like it could  be a good idea to unify the effects UI. The variability between skins now is  awfully confusing and unintuitive.

ferranlala committed:
Merge remote-tracking branch 'upstream/1.12' into skin-layout-revamp-dev

Conflicts:
	res/skins/LateNight/mixer.xml

ferranpujolcamins commented:
> The only bug I see is that In the larger sizes, the library button is clickable but doesn't do anything.

It's a workaround for another bug (worse in my opinion). As I commented before:

ferranpujolcamins commented:
Oh, I've just thought I could change the color or the "L" and "R" of the crossfader assign button to red and blue, it's a quick fix. The user could see at a glance where is a deck assigned. Would you like that?

ferranpujolcamins commented:
Another possible quick fix: can I get rid of the striped backgrounds on the decks and samplers? I don't really like them .

ywwg commented:
> The user could see at a glance where is a deck assigned. Would you like that?

leave that alone for now, we use red and blue for deck 1/2 vs 3/4 so it would be confusing to also use red/blue for right/left.  Or just make both the R and L yellow text -- there's no real need to color code.

Please put it back so the library button is not clickable -- it makes more sense than to have the button be broken.  Or figure out a way to skin it so it is always on but has a "disabled" appearance.  Like, flat with grayer text.

ferranpujolcamins commented:
> Please put it back so the library button is not clickable -- it makes more sense than to have the button be broken. Or figure out a way to skin it so it is always on but has a "disabled" (unclickable) appearance. Like, flat with grayer text.

I can do that we will still have the same problem with the view menu check option. I think this can't completely be solved until we have a way to communicate to Mixxx which CO a skin offers.

ywwg commented:
how is it that in Deere, the menu items are grayed out?

> can I get rid of the striped backgrounds on the decks and samplers

no thanks :).  Maybe in the future I'll change my mind but let's leave them for now.

ferranpujolcamins commented:
> how is it that in Deere, the menu items are grayed out?

I see all of them not grayed out. I'm investigating why the CO not supported are still active.

Be-ing commented:
How is progress on this?

ferranpujolcamins commented:
Hi, I thing the only thing that still needs to be done is this:


> Please put it back so the library button is not clickable -- it makes more sense than to have the button be broken. Or figure out a way to skin it so it is always on but has a "disabled" (unclickable) appearance. Like, flat with grayer text.

I'll try to fix this tomorrow.

ferranlala committed:
LateNight: hide the library button when its useless.

ferranpujolcamins commented:
Done. What else is blocking this?

Be-ing commented:
Could you make the crossfader assignment CO persistent? As an example use case, I may want to keep decks 1 & 2 assigned to left and right, but not assign 3 & 4 to the crossfader.

Be-ing commented:
I am opposed to merging this unless https://bugs.launchpad.net/mixxx/+bug/1462061 is fixed. Without the ability to hide the library, that bug can't be worked around.

Be-ing commented:
This is still a bit too tall for my screen 1366x768 laptop screen. :/
![late-night-pr-screenshot](https://cloud.githubusercontent.com/assets/9455094/8885596/d3dd161c-3226-11e5-8be1-257ef195a248.png)

ferranpujolcamins commented:
Hi @Be-ing , Shrinking the height of LateNight is yet another thing I'm planning to do.
I'm investigating the bug you've mentioned as well.

daschuer commented:
What is current state of this? I am afraid the git branch is messed up since there are unrelated commits listed, and it is not in a merge-able state. Is it still a 1.12 candidate?

ywwg commented:
> Is it still a 1.12 candidate?

No, let's keep this master-only.  I think LateNight is in fine shape for 1.12.

ywwg commented:
(to be specific, the code changes and changes to other skins required by this alteration, coupled with the "nice-to-have" nature of the changes tip it in my mind to something for master.  Although the height issue in latenight is annoying, it's not really a blocker)

ferranpujolcamins commented:
In fact this doesn't solve the issue of LateNight being too tall anyway.
It's true that this was a first step towards a solution (hide/show
crossfader), but it has mutated to a layout options re-factoring.

We still need to make a lot of work in order to offer the user a *complete*
("all" functions available in each skin), *consistent* (skins can't have a
very different work-flow for common tasks such as beat-looping, fx, or the
layout options discussed in this PR), *polished* (that endless list of
little things that can look and work better) and *beautiful* experience
with skins. Don't worry we will eventually make it for 1.13 or whatever
release.

This is going to be long, so let's just take a compromise on each of the
skin aspects above for now. Let's just try to give birth to 1.12 ASAP :)

2015-08-03 14:46 GMT+02:00 Owen Williams <notifications@github.com>:

> (to be specific, the code changes and changes to other skins required by
> this alteration, coupled with the "nice-to-have" nature of the changes tip
> it in my mind to something for master. Although the height issue in
> latenight is annoying, it's not really a blocker)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/639#issuecomment-127220351>.
>

Be-ing commented:
@ferranpujolcamins, are you planning on finishing this PR for 2.1?

ferranpujolcamins commented:
Sure, I have some free days. I'll take a look.
El dia 19/01/2016 4:58 a. m., "Be" <notifications@github.com> va escriure:

> @ferranpujolcamins <https://github.com/ferranpujolcamins>, are you
> planning on finishing this PR for 2.1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/639#issuecomment-172731060>.
>

ferranpujolcamins commented:
Well, actually not. Sorry.

2016-01-19 8:22 GMT+01:00 Ferran Pujol Camins <ferranpujolcamins@gmail.com>:

> Sure, I have some free days. I'll take a look.
> El dia 19/01/2016 4:58 a. m., "Be" <notifications@github.com> va escriure:
>
>> @ferranpujolcamins <https://github.com/ferranpujolcamins>, are you
>> planning on finishing this PR for 2.1?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mixxxdj/mixxx/pull/639#issuecomment-172731060>.
>>
>

ferranpujolcamins commented:
Superseded by https://github.com/mixxxdj/mixxx/pull/913, which merges to master cleanly.
